pulsar-functions/instance/src/main/python/python_instance_main.py

```diff
@@ -74,6 +73,15 @@ def main():
   args = parser.parse_args()
   function_details = Function_pb2.FunctionDetails()
   json_format.Parse(args.function_details, function_details)
+
+  if function_details.runtime == Function_pb2.FunctionDetails.Runtime.Value("PYTHON_WHEEL"):
+    try:
+      util.install_wheel(str(args.py))
```

Comment:
I am not familiar with python. but isn't that manipulating the environment, and doesn't it require sudos and such?

Reply:
I've redone that part with installing using virtualenv to minimize impact on the system and needing permissions